persistence/mongo/src/main/scala/za/co/absa/spline/persistence/mongo/dao/LineageDAOv4.scala

```diff
@@ -51,6 +52,10 @@ class LineageDAOv4(override val connection: MongoConnection) extends BaselineLin
     super.save(lineage)
   }
 
+  override def saveProgress(event: ProgressDBObject)(implicit e: ExecutionContext): Future[Unit] = {
```

Comment:
will add dummy throwing exception in `BaseDao` and override that in `DaoV5` with above. 

Adding yet another interface is not a good idea. It may have been better to not to have MultiVersionDao at all and instead call from `MongoReader` n `MongoWriter` always directly `latestDao`, which would be responsible for calling the previous one. And so on. (daov5 would call daov4 then dao4 would call dao3).

But it is not worth changing now.

Reply:
Agree.
Regarding v5 to call v4 and so on I partially agree. API would be much clearer and separated, yes. But on the other side you'd have hard dependency between DAOs (v5 couldn't work without v4 etc) and each DAO would have to be responsible for upgrading the response from DAO vN-1 and combining the results. Though it would be possible to split the responsibility by implementing a chain-of-responsibility pattern, but I'm not sure if the final API would be clearer than the current one in the result.